Userland/Libraries/LibWeb/HTML/HTMLMetaElement.cpp

```diff
@@ -41,6 +46,43 @@ void HTMLMetaElement::inserted()
 {
     Base::inserted();
 
+    // https://html.spec.whatwg.org/multipage/semantics.html#meta-theme-color
+    // 1. To obtain a page's theme color, user agents must run the following steps:
+    //     * The element is in a document tree
+    //     * The element has a name attribute, whose value is an ASCII case-insensitive match for theme-color
+    //     * The element has a content attribute
+    auto name = deprecated_attribute(AttributeNames::name);
```

Comment:
It would be nice if we could use `attribute(AttributeNames::name);` here instead of the deprecated version of this function.

Reply:
I suppose it would be neat, however, we'd have to deal with fallibility in that case since `String::trim()` may fail, but `DeprecatedString::trim()` may not. `attribute()` is also implemented in terms of `deprecated_attribute()`, so not entirely sure if it'd make that big of a difference. 🤷

Reply:
`DeprecatedString::trim()` may also fail in the same way as `String::trim()` in that it may get an OOM, DeprecatedString just hides that and will crash itself. In LibWeb, we don't care about allocation failure and are happy to just crash the process if that happens. So we can just `MUST` the result here. Definitely a pitfall of using new String in that it's more verbose because of that.

And yup, you're right that it's currently implemented in terms of the other. But at some stage that will change. I would just like to stop adding more cases of the old API where it is possible / not too bad to do so - especially since this is making use of the null state of DeprecatedString. Don't feel super strongly though, just think it will save future work :^)

Reply:
Sure. Changed it to use `attribute`, but managed to keep it infallible by converting the `String` to `StringView` before trimming the whitespace.